crates/matrix-sdk/src/media.rs

```diff
@@ -198,49 +203,51 @@ impl Media {
 
         let inferred_extension = mime2ext::mime2ext(content_type);
 
-        let body_path = body.as_ref().map(Path::new);
-        let filename = body_path.and_then(|f| f.file_name().and_then(|f| f.to_str()));
-        let filename_with_extension = body_path.and_then(|f| {
+        let filename_as_path = filename.as_ref().map(Path::new);
+        let filename_without_extension =
+            filename_as_path.and_then(|f| f.file_name().and_then(|f| f.to_str()));
```

Comment:
Maybe rename this first `|f|` to `|path|` so it's clear we get just the file name and not the extension because we're asking that from the  `Path`?

Reply:
drive-by nit: use an external `.and_then` instead of putting it inside the first `and_then`:

```suggestion
            filename_as_path.and_then(|f| f.file_name()).and_then(|f| f.to_str());
```

Reply:
Ah yep makes sense. Also looking at this code, renaming `filename` to `filename_without_extension` is wrong. Both filenames are the same if there's an extension and if there isn't then `filename_with_extension` is `None`.

I've updated the code to use a bool `filename_has_extension` and not a second copy of the string.